### ISSUE TITLE
Single line function declaration after checking function_exists incorrectly detected as multiline function declaration

### ISSUE THREAD

**Issue by sdon2** [squizlabs/PHP_CodeSniffer#3682]
**Incorrect function type detection**
Single line function declaration after checking function_exists incorrectly detected as multiline function declaration

**Code sample**
```php
if (!function_exists('get_setting')) {
    function get_setting($name, $default = null) {
        return \App\Models\Setting::where('name', $name)->first()->value ?? $default;
    }
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset>
    <rule ref="PSR2" />
    <file>app</file>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
FILE: app\Helpers\functions.php
----------------------------------------------------------------------
FOUND 1 ERRORS AFFECTING 1 LINES
----------------------------------------------------------------------
  4 | ERROR | [x] Opening brace should be on a new line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Expected behavior**
There should be no errors.

**Versions:**
 - OS: Windows 10
 - PHP: 7.4
 - PHPCS: 3.7.1 (stable)
 - Standard: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine

**Options tried:**
I know there is an option to disable these checks:
```xml
<exclude name="Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine" />
```
However I need to check **_function declaration braces must not be on newline_**.

**Comment by gsherwood** [comment]
The function is a multi-line function declaration because it crosses 3 lines of code. A single-line function declaration is when the opening and closing braces are on the same line, like this:
```php
function get_setting($name, $default = null) { return true; }
```

But that really doesn't matter because the sniff is there to enforce that all functions must be formatted as multi-line declarations as per PSR2, which requires all functions to have their opening brace on a new line.

It looks like everything is working correctly to me.

**Comment by sdon2** [comment]
> The function is a multi-line function declaration because it crosses 3 lines of code.

Can I override this?

**Comment by sdon2** [comment]
Thanks for your help. I understood. If I override the single line function standard, all of my functions should be like:

```
if (!function_exists('get_setting')) {
    function get_setting($name, $default = null) {
        return \App\Models\Setting::where('name', $name)->first()->value ?? $default;
    }
}

function save_setting($name, $value) {
    $setting = \App\Models\Setting::where('name', $name)->first();
    if ($setting) {
        $setting->update(['value' => $value]);
    }
}
```
This is against standards, and I don't like it too. I'll stay with default rules. Thanks again.
